### ISSUE TITLE
SAML SSO with Azure AD

### ISSUE THREAD

**Issue by skinnyxpuppy** [argoproj/argo-cd#7896]
Hello guys, 
Apologies if this is the wrong location for the post - I have actually tried joining the Slack community to ask, but the slack link on the support page is stating the URL is no longer active. 

I am currently following the steps to configure Azure AD SAML using the provided documentation
 https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/microsoft/
After setting up, I am successfully getting the "Log in via SAML" selection on the landing page, but after an auth I get with O36, I get the following error. 

Failed to authenticate: no attribute with name "Group": [http://schemas.microsoft.com/identity/claims/tenantid http://schemas.microsoft.com/identity/claims/objectidentifier http://schemas.microsoft.com/claims/groups.link http://schemas.microsoft.com/identity/claims/identityprovider http://schemas.microsoft.com/claims/authnmethodsreferences http://schemas.microsoft.com/ws/2008/06/identity/claims/wids email]

I have successfully set up the claims as requested, but I feel like I have something slightly wrong here. Thanks for the help! 

![ArgoCD](https://user-images.githubusercontent.com/95872294/145446305-2c1bc8ef-176a-4975-b4a5-11093b8d8576.JPG)
![ArgoCD2](https://user-images.githubusercontent.com/95872294/145446310-15cc49fc-91b7-4142-a6b1-a8f16964201f.JPG)
![ArgoCd3](https://user-images.githubusercontent.com/95872294/145446312-307d75df-c1c8-47be-8a8d-a65f2f3c221b.JPG)
![argocd4](https://user-images.githubusercontent.com/95872294/145446314-c8e17345-d09f-4664-be26-733798a2ba2a.JPG)

**Comment by skinnyxpuppy** [comment]
This can be closed, I have resolved the issue. 

For some reason, when I used "all groups" it does not work, but it works fine if I select "groups assigned to this application"

**Comment by a3gill** [comment]
It probably caused a "token overage"

tl;dr: Too many groups cause token to exceed max length
